### ISSUE TITLE
Incompatible with Ejie Clover

### ISSUE THREAD

**Issue by Darthagnon** [xupefei/QuickLook#213]
Ejie Clover is a Windows Explorer replacement/shell plugin (I'm not sure which) that add convenient Chrome-style tabs to Windows Explorer (Website: [http://en.ejie.me/](urll)). It is very useful; it's just a pity that QuickLook (also amazing and very useful) doesn't work with Clover. Would it be possible to make QuickLook compatible with Clover as a 3rd-party file manager?

**Comment by Darthagnon** [comment]
Noticed that this was added to the 0.3.3 milestone; was wondering, does that mean it's currently compatible with Clover, since the current version is 0.3.5?

**Comment by xupefei** [comment]
Unfortunately no. The current version 0.3.5 means nothing but minor tweaks
from 0.3.3. Because Windows Store does not accept versions like "0.3.3.1".
You can see the changes of version 0.3.4 and 0.3.5 on the release page.

**Comment by Rictusempra** [comment]
Just a little question here
As how I remember, Clover should be Explorer hook, which shouldn't interfere with QuickLook at all.

**Comment by xupefei** [comment]
It may have changed the class name of Explorer window. QL checks a window's class name before further processing in QuickLook.Native dlls.

cheers,
Paddy

**Comment by Darthagnon** [comment]
Thank you! 
(PS It's an awesome tool)

**Comment by xupefei** [comment]
I have just tried Clover 3.4.3. QL works without any problem. Can you try again?

**Comment by multiplatformsgamer** [comment]
clover 3.4.3 依旧不支持。v.0.3.5

**Comment by xupefei** [comment]
Clover 3.4.3 works with QL Store version my Windows 10.
What OS are you using?

cheers,
Paddy

**Comment by Darthagnon** [comment]
I just tried Clover and Quicklook (I had previously noticed the problem on my brother's Win10 computer, where he was trying the Windows Store version of Quicklook and Clover). I used Quicklook 0.3.5 msi version (Unrelated bug: for some reason, the portable version didn't want to work for me, and the msi version even had a problem, saying "unable to create startup shortcut", and making a 0kb startup shortcut) and the latest version of Clover (3.4.3.04081). Quicklook successfully previews files on the Windows desktop, but nothing happens when I try to Quicklook with Clover. 

My PC is a Win7 Home Premium x64 2010 Core i5 4GB RAM HP Pavilion.

**Comment by Darthagnon** [comment]
I'm running Windows 7, and sounds like the other poster was running Windows 7, too.  Maybe there's a bug there? Though it didn't work on my brother's Win10 computer... I'll see if I can convince him to install it again. Any particular version?

**Comment by Darthagnon** [comment]
I'm currently trying Seer 0.8.1 from https://sourceforge.net/projects/ccseer/. It is compatible with Clover, and uses less RAM than QuickLook. NB I still prefer QuickLook, it's much prettier.
------------------------------------------------------------------------------------------------------------------
Seer is on SourceForge, though I can't tell if it's open source (can't see the source code anywhere). If it's available, perhaps that would show a method to implement Clover compatibility in QuickLook?

**Comment by xupefei** [comment]
As said before, I can use QL with Clover. This issue is not reproducable to
me.

Please provide details steps about how to make it happens, otherwise I cant
help.

**Comment by xupefei** [comment]
Well, I have just recalled that all such issues happens on Windows 7. Let me get a VM and try.

Cheers,
Paddy

**Comment by xupefei** [comment]
**TL;DR: This is a bug on Windows 7 with Ejie Clover installed. It is caused by Clover. QuickLook will not fix it. You can use Seer mentioned by @Darthagnon.**

============================

Today I tried to use a VM Win7 with Clover. QL does not work. After some investigations, I have found that Clover did something bad and breaks the COM call from QL when obtaining the selected file. 

The reproducible environment is:
```
Windows 7 32 or 64 bit with Ejie Clover installed
```
Since Clover is not open-source, I have no idea how to modify QL to make them compatible. Sorry.

> I'm currently trying Seer 0.8.1. It is compatible with Clover.

Seer is using a slightly different COM routine so it works. However, I will not modify QL's routine since Seers's way is ugly and inefficient. Moreover, this issue has limited affect since only few people are still using Windows 7 in these days.

**Comment by lurui45** [comment]
After viewing all your discussion, it seems that I need to do a difficult choice between clover and QL (due to some reason, I need to use win7 ), but apparently, I will discard clover :(,  clover as a china software, it integrates  a lot of stealthy advertisements, I don't know if you guys meet these ad caused by clover. such as a pop window in bottom-right, or adding links in your explorer tabs... it's disgusting.

**Comment by Simon-Ace** [comment]
@lurui45  If you want to use clover again, I recommend you to use previous version of clover such as v3.0.406. This version doesn't have any advertisements.

**Comment by Darthagnon** [comment]
Just wanted to report that the issue is still present (Win7 x64 Home Premium, Quicklook 3.6.5 MSI) , despite updates to Clover (now at v3.5.0, I recommend the [lrepacks.ru](https://lrepacks.ru/repaki-sistemnyh-programm/385-clover-repack-amp-portable.html#) version, as it removes the chinese ad elements and implements an English un/installer)

Would a more generic  (or customisable?) trigger hook for Quicklook be a possible solution

**Comment by Darthagnon** [comment]
[Window Tabs](https://github.com/leafOfTree/WindowTabs/releases) is now FOSS, and it looks and works in a very similar way to Clover. With the added bonuses:
1, It works with QuickLook
2. It (optionally) applies tabs to all software, kinda like [TidyTabs](https://www.nurgo-software.com/products/tidytabs)
